client/v2/client.go

```diff
@@ -413,16 +414,45 @@ func (uc *udpclient) Write(bp BatchPoints) error {
 	for _, p := range bp.Points() {
 		pointstring := p.pt.RoundedString(d) + "\n"
 
-		// Write and reset the buffer if we reach the max size
 		if b.Len()+len(pointstring) >= uc.payloadSize {
 			if _, err := uc.conn.Write(b.Bytes()); err != nil {
 				return err
 			}
 			b.Reset()
 		}
 
-		if _, err := b.WriteString(pointstring); err != nil {
-			return err
+		if b.Len()+len(pointstring) <= uc.payloadSize {
+			if _, err := b.WriteString(pointstring); err != nil {
```

Comment:
There is no need to check `err`, it will always be `nil`: https://golang.org/pkg/bytes/#Buffer.WriteString

It feels weird, but we know we're dealing with a `bytes.Buffer` here.